Dockerfile

```diff
@@ -1,149 +1,186 @@
+        rm -rf ${EMBREE_INSTALL_DIR}/build ${EMBREE_INSTALL_DIR}/embree ; \
+        # Clone and install MOAB
+        mkdir -p $HOME/MOAB && cd $HOME/MOAB ; \
+        git clone  --single-branch -b ${MOAB_BRANCH} --depth 1 ${MOAB_REPO} ; \
+        mkdir build && cd build ; \
+        cmake ../moab \
+                    -DENABLE_HDF5=ON \
+                    -DENABLE_NETCDF=ON \
+                    -DBUILD_SHARED_LIBS=OFF \
+                    -DENABLE_FORTRAN=OFF \
+                    -DENABLE_BLASLAPACK=OFF ; \
+        make 2>/dev/null -j${compile_cores} install ; \
+        cmake ../moab \
+                    -DENABLE_PYMOAB=ON \
+                    -DBUILD_SHARED_LIBS=ON \
+        make 2>/dev/null -j${compile_cores} install ; \
+        cd pymoab && bash install.sh; \
+        python setup.py install; \
+        rm -rf $HOME/MOAB ; \
+        # Clone and install Double-Down
+        mkdir -p $HOME/Double_down && cd $HOME/Double_down; \
+        git clone --single-branch -b ${DD_BRANCH} --depth 1 ${DD_REPO} ; \
+        mkdir build && cd build; \
+        cmake ../double-down \
+                    -DCMAKE_INSTALL_PREFIX=${DD_INSTALL_DIR} \
+                    -DMOAB_DIR=/usr/local \
+                    -DEMBREE_DIR=${EMBREE_INSTALL_DIR} ; \
+        make 2>/dev/null -j${compile_cores} install ; \
+        rm -rf ${DD_INSTALL_DIR}/build && ${DD_INSTALL_DIR}/double-down; \
+        # Clone and install DAGMC
+        mkdir -p $HOME/DAGMC && cd $HOME/DAGMC; \
+        git clone --single-branch -b ${DAGMC_TAG} --depth 1 ${DAGMC_REPO} ; \
+        mkdir build && cd build ; \
+        cmake ../DAGMC \
+                    -DBUILD_TALLY=ON \
+                    -DCMAKE_INSTALL_PREFIX=${DAGMC_INSTALL_DIR} \
+                    -DMOAB_DIR=/usr/local \
+                    -DDOUBLE_DOWN=ON \
+                    -DDOUBLE_DOWN_DIR=${DD_INSTALL_DIR} \
+                    -DCMAKE_PREFIX_PATH=${DD_INSTALL_DIR}/lib \
+                    -DBUILD_STATIC_LIBS=OFF \
+                    -DBUILD_STATIC_EXE=OFF ; \
+        make 2>/dev/null -j${compile_cores} install ; \
+        rm -rf ${DAGMC_INSTALL_DIR}/DAGMC ${DAGMC_INSTALL_DIR}/build ; \
+        # Clone and install OpenMC with DAGMC support
+        mkdir -p $HOME/OpenMC && cd $HOME/OpenMC; \
+        git clone --shallow-submodules --recurse-submodules -b ${OPENMC_BRANCH} --depth=1 ${OPENMC_REPO} ; \
+        mkdir build && cd build ; \
+        cmake ../openmc \
+                    -Doptimize=on \
+                    -Ddagmc=ON \
+                    -Ddebug=on \
+                    -DDAGMC_DIR=${DAGMC_INSTALL_DIR} \
+                    -DHDF5_PREFER_PARALLEL=on ; \
+        make 2>/dev/null -j${compile_cores} install ; \
+        cd ../openmc && pip install -e .[test] ; \
+    elif [ "$include_libmesh" = "true" ]; then \
+        # Install addition packages required for LIBMESH
+        apt-get -y install m4 libnetcdf-dev libpnetcdf-dev ; \
```

Comment:
```suggestion
        cd ../openmc && pip install -e .[test]

RUN if [ "$include_libmesh" = "true" ]; then \
        # Install addition packages required for LIBMESH
        apt-get -y install m4 libnetcdf-dev libpnetcdf-dev ; \
```

Reply:
What do you think @AI-Pranto this breaks it down into two Docker layers to cover the without dagmc but with libmesh case

Reply:
oh wait on second thoughts, I see that we have a few things going on here. Ignore my earlier comment. But I think we don't have all the options covered here

|          | libmesh | dagmc |
|----------|---------|-------|
| option 1 | yes     | yes   |
| option 2 | yes     | no    |
| option 3 | no      | yes   |
| option 4 | no      | no    |

Reply:
How about we use some ```&&```  symbols in the if statements
```
if [ "$include_dagmc" = "true" ] && [ "$include_libmesh" = "true" ]; then
   ...
if [ "$include_dagmc" = "true" ] && [ "$include_libmesh" = "false" ]; then
   ...
if [ "$include_dagmc" = "false" ] && [ "$include_libmesh" = "false" ]; then
   ...
if [ "$include_dagmc" = "false" ] && [ "$include_libmesh" = "false" ]; then
```

Reply:
Hrmm yeah that's going to create a lot of branches if we continue with the current design. Is there a way to factor our the OpenMC build section at least?

Reply:
We can revert some of the docker layer compression, if that is ok with @AI-Pranto

Reply:
@Shimwell No problem, that'll be great to cover all possible combinations